sdks/python/setup.py

```diff
@@ -131,6 +131,10 @@ def get_version():
     # Avro 1.9.2 for python3 was broken. The issue was fixed in version 1.9.2.1
     'avro-python3>=1.8.1,!=1.9.2,<1.10.0',
     'crcmod>=1.7,<2.0',
+    # dataclasses backport for Python 3.6. No version boundary so that Python
+    # picks the correct one based on its version. For more details, see
+    # https://pypi.org/project/dataclasses/
+    'dataclasses',
```

Comment:
I don't see anything at https://pypi.org/project/dataclasses/ explaining why we don't need a version bound, could you clarify?

It could make sense to add a `python_version<3.7`, so that if/when we drop python 3.6 support it's clear we can drop this.

Reply:
Yes it makes sense. I'll add a `python_version<3.7` TODO to remove it when we drop Python 3.6.

**More context on why no version bound**:

Based on https://pypi.org/project/dataclasses/

>this repo should only be used for historical purposes

If you look at the "Release History" section of it, the newer version of the package is for backport of Python 3.6.
For Python 3.7, the compatible version is actually lower (though installation is not needed at all).

Tested as something like: 

| Python version | Compatible dataclasses version |
| --------------- | -------------------------------- |
|3.7 | 0.5|
|3.8 | 0.6|

A range of versions doesn't make sense since each Python version uses a specific version of dataclasses.

Reply:
What I mean was to put a `python_version<3.7` on the requirement, that way it will just be ignored entirely for newer versions (and it will also remind us to drop the dep in the future).

Reply:
It looks like we don't actually have any examples of this in our setup.py to draw from anymore! We used to do it all over the place because of py2.7 and 3.5. Here is an example of what I'm talking about: https://stackoverflow.com/a/32643122/2952954

Reply:
```suggestion
    'dataclasses;python_version<"3.7"',
```
WDYT about making it explicit like this?

Reply:
OMG, that's exactly what I wanted! Thanks, I'll make the change.